unix/modsocket: Add "sendall" method to socket class.

agatti opened the pull request:
### Summary

This PR adds an implementation for "socket.sendall" to the Unix port.

Right now the implementation is a wrapper over a single "socket.send" call, since it wasn't possible to let "socket.send" perform a partial transfer.  With no partial transfers it is not possible to have a testable implementation following the expected behaviour, so a single send operation is done and OSErr(EINTR) is raised if a partial transfer occurs.

The discussion for the issue linked to this PR contains more information about the efforts made to let partial transfers happen.

This fixes #16803.

### Testing

A new test is introduced to exercise `socket.sendall`, called `tests/multi_net/tcp_sendall.py`.  This could have been folded into `tests/multi_net/tcp_data.py`, but the Zephyr port doesn't have a `socket.sendall` implementation.

### Trade-offs and Alternatives

There's a modest size increase.  The footprint increase could have been smaller if `socket.sendall` was aliased to `socket.send` like the CC3200 port does, but then the behaviour would be slightly incompatible with CPython (`socket.sendall` is supposed to return `None`, not the sent bytes count).

dpgeorge reviewed (commented):
Thanks for opening this PR.  If we can get the code size down, that would be great (see my suggestion).

agatti committed:
unix/modsocket: Add "sendall" method to socket class.

This commit adds an implementation for "socket.sendall" to the Unix
port.

Right now the implementation is a wrapper over a single "socket.send"
call, since it wasn't possible to let "socket.send" perform a partial
transfer.  With no partial transfers it is not possible to have a
testable implementation following the expected behaviour, so a single
send operation is done and OSErr(EINTR) is raised if a partial transfer
occurs.

The discussion for the issue linked to this PR contains more information
about the efforts made to let partial transfers happen.

This fixes #16803.

Signed-off-by: Alessandro Gatti <a.gatti@frob.it>

dpgeorge reviewed (approved):
Thanks for updating.

Tested on PYBD_SF6 <-> unix and the new test passes.
